### DISCUSSION TITLE
Bootstrap options using 2 ports (web and internal API)  (edgex-ui-go)

### DISCUSSION THREAD

**Discussion by mglmc** [edgexfoundry/community#236] [Q&A] [answered]
Hello,

**TLDR**: How to have 2 ports at the same time using Bootstrap in edgex-ui-go, one for the internal API and the other one for serving the web

We noticed that the **edgex-ui-go** serves the web in a specific port that is not the default one (80) for web clients.
Additionally, [here](https://github.com/edgexfoundry/edgex-ui-go/blob/main/internal/router.go) is serving the web in the same port that the internal API. That is something that usually should be doing it using another port for security concerns

We're trying to have 2 ports. One external (80) for serving the web, and, another one (the default's) for the internal API, and, do the conversion using the reverse proxy as it is doing it for the rest of the services.

The problem here is that Bootstrap manages it and it applies an abstraction layer. 

In the docs we only see that there is an option for the IP server [ServerBindAddr](https://docs.edgexfoundry.org/3.1/design/legacy-requirements/device-service/#service-section:~:text=monitor%20its%20liveliness.-,ServerBindAddr,-String)

How can we have 2 ports using in edgex-ui-go?

**Comment by lenny-goodell** [comment, marked as answer]
The best way to handle  this would be to add second web server running just the Web portion. This would allow for a second port.
The one in bootstrap is for internal APIs.

**Reply by mglmc** [reply]
Thank you for the response!

**Comment by cloudxxx8** [comment]
close as answered
